test/e2e/crd_e2e_test.go

```diff
@@ -0,0 +1,194 @@
+			TypeMeta: metav1.TypeMeta{APIVersion: "apiextensions.k8s.io/v1beta1", Kind: "CustomResourceDefinition"},
+			ObjectMeta: metav1.ObjectMeta{
+				Name: crdNamePlural + "." + crdGroup,
+			},
+			Spec: apiextensionsv1beta1.CustomResourceDefinitionSpec{
+				Group:   crdGroup,
+				Version: "v1",
+				Names: apiextensionsv1beta1.CustomResourceDefinitionNames{
+					Singular: crdName,
+					Plural:   crdNamePlural,
+					Kind:     "test",
+				},
+			},
+		}
+
+		// create v1beta1 CRD on server
+		oldcrd, err := c.ApiextensionsInterface().ApiextensionsV1beta1().CustomResourceDefinitions().Create(context.TODO(), oldv1beta1CRD, metav1.CreateOptions{})
+		Expect(err).ToNot(HaveOccurred())
+		By("created CRD")
+
+		// poll for CRD to be ready (using the v1 client)
+		Eventually(func() (bool, error) {
+			fetchedCRD, err := c.ApiextensionsInterface().ApiextensionsV1().CustomResourceDefinitions().Get(context.TODO(), oldcrd.GetName(), metav1.GetOptions{})
+			if err != nil || fetchedCRD == nil {
+				return false, err
+			}
+			return checkCRD(fetchedCRD), nil
+		}, 5*time.Minute, 10*time.Second).Should(Equal(true))
+
+		oldCRDConvertedToV1, err := c.ApiextensionsInterface().ApiextensionsV1().CustomResourceDefinitions().Get(context.TODO(), oldcrd.GetName(), metav1.GetOptions{})
+		Expect(err).ToNot(HaveOccurred())
+
+		// confirm the v1 crd as is as expected
+		// run ensureCRDV1Versions on results
+		newCRD := &apiextensionsv1.CustomResourceDefinition{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: crdNamePlural + "." + crdGroup,
+			},
+			Spec: apiextensionsv1.CustomResourceDefinitionSpec{
+				Group: crdGroup,
+				Versions: []apiextensionsv1.CustomResourceDefinitionVersion{
+					{
+						Name:   "v1",
+						Served: true,
+					},
+				},
+				Names: apiextensionsv1.CustomResourceDefinitionNames{
+					Singular: crdName,
+					Plural:   crdNamePlural,
+				},
+			},
+		}
+
+		err = catalog.EnsureV1CRDVersions(oldCRDConvertedToV1, newCRD)
+		Expect(err).ToNot(HaveOccurred())
+	})
+	It("creates v1beta1 crds with a v1beta1 schema successfully", func() {
+		By("This test proves that OLM is able to handle v1beta1 CRDs successfully. Creating v1 CRDs has more " +
+			"restrictions around the schema. v1beta1 validation schemas are not necessarily valid in v1. " +
+			"OLM should support both v1beta1 and v1 CRDs")
+		c := newKubeClient(GinkgoT())
+		crc := newCRClient(GinkgoT())
+		defer func() {
+			require.NoError(GinkgoT(), crc.OperatorsV1alpha1().Subscriptions(testNamespace).DeleteCollection(context.TODO(), metav1.DeleteOptions{}, metav1.ListOptions{}))
+		}()
+
+		mainPackageName := genName("nginx-update2-")
+		mainPackageStable := fmt.Sprintf("%s-stable", mainPackageName)
+		stableChannel := "stable"
+		mainNamedStrategy := newNginxInstallStrategy(genName("dep-"), nil, nil)
+
+		crdPlural := genName("ins-v1beta1-")
+		crdName := crdPlural + ".cluster.com"
+		mainCRD := apiextensions.CustomResourceDefinition{
```

Comment:
shouldn't this be a v1beta1 instead of internal?

Reply:
all of our e2e tests currently create an internal crd which gets converted to a v1beta1 later on when creating the CSV. I could create the v1beta1 directly but would then need to write my own `createCSV` and other helper functions

Reply:
TODO: we should update all our e2e to create v1beta1 CRDs and update the createCSV to accept the versioned CRD type